tests/app/questionnaire/test_questionnaire_store_updater.py

```diff
@@ -497,6 +497,156 @@ def test_update_answers_captures_answer_dependencies(
+                        "value": "answer updated",
+                        "list_item_id": None,
+                    },
+                    {
+                        "answer_id": "second-answer",
+                        "value": "second answer",
+                        "list_item_id": None,
+                    },
+                ]
+            ),
+        ),
+        (  # When the answer dependent has an answer_id, but the answer dependency value is not changed, then the answer store should not change
+            "second-answer",
+            "original answer",
+            AnswerStore(
+                [
+                    {
+                        "answer_id": "first-answer",
+                        "value": "original answer",
+                        "list_item_id": None,
+                    },
+                    {
+                        "answer_id": "second-answer",
+                        "value": "second answer",
+                        "list_item_id": None,
+                    },
+                ]
+            ),
+        ),
+    ],
+)
+def test_update_answers_with_answer_dependents(
+    mock_schema, answer_dependent_answer_id, updated_answer_value, expected_output
+):
+    answer_store = AnswerStore()
+    answer_store.add_or_update(
+        Answer(answer_id="first-answer", value="original answer")
+    )
+    answer_store.add_or_update(Answer(answer_id="second-answer", value="second answer"))
+
+    mock_schema.get_answer_ids_for_question.return_value = ["first-answer"]
+    mock_schema.answer_dependencies = {
+        "first-answer": {
+            AnswerDependent(
+                section_id="section",
+                block_id="second-block",
+                for_list=None,
+                answer_id=answer_dependent_answer_id,
+            )
+        },
+    }
+
+    form_data = MultiDict({"first-answer": updated_answer_value})
+
+    location = Location(
+        section_id="section",
+        block_id="first-block",
+    )
+    current_question = mock_schema.get_block(location.block_id)["question"]
+    questionnaire_store_updater = get_questionnaire_store_updater(
+        schema=mock_schema,
+        answer_store=answer_store,
+        list_store=ListStore(),
+        current_location=location,
+        current_question=current_question,
+    )
+    questionnaire_store_updater.update_answers(form_data)
+
+    assert answer_store == expected_output
+
+
+def test_update_repeating_answers_with_answer_dependents(mock_schema):
+    # Given repeating dependent answers
+    answer_store = AnswerStore()
+    answer_store.add_or_update(
+        Answer(answer_id="first-answer", value="original answer")
+    )
+    answer_store.add_or_update(
+        Answer(answer_id="second-answer", value="second answer", list_item_id="abc123")
+    )
+    answer_store.add_or_update(
+        Answer(answer_id="second-answer", value="second answer", list_item_id="xyz456")
+    )
```

Comment:
Minor, but I think it would be clearer if the store was built inline. The `add_or_update` is only needed, if we are doing something dynamic. Same for the other test.
```suggestion
    answer_store = AnswerStore(
        [
            AnswerDict(answer_id="first-answer", value="original-answer"),
            AnswerDict(
                answer_id="second-answer", value="second answer", list_item_id="abc123"
            ),
            AnswerDict(
                answer_id="second-answer", value="second answer", list_item_id="xyz456"
            ),
        ]
    )
    )
```